### DISCUSSION TITLE
Efficient way to get multiple game stats for a user? (CMsgClientGetUserStats or equivalent)

### DISCUSSION THREAD

**Discussion by ABaumher** [SteamRE/SteamKit#1222] [General]
The code i'm looking at sends multiple `CMsgClientGetUserStats`, one for each game in the user's catalog. The goal is to get achievements, play time, etc. It works, but if the user has 1000+ games, that's a ton of messages sent to back and forth, and suddenly we have 1000+ tasks to wait on. We can simply start them async and await them after everything else finishes so we aren't bottlenecked by all these results, but is there a smarter/more efficient way?

One of our developers had the clever idea of sending a large "multi" message that packs these together so we don't have a whole mess of them, like the ones Steam sends us on occasion. Is this possible, and if so, would it be worth it?

**Comment by yaakov-h** [comment]
Last I checked, you can only receive a Multi message, you can't send one - i.e. if you send one, the servers will ignore it.

(I know, because I tried sending a zip bomb to the servers once.)
